config/utils.ts

```diff
@@ -0,0 +1,447 @@
+    process.env[ENVIRONMENT_VARIABLES.HUBSPOT_ACCOUNT_ID] ||
+    process.env[ENVIRONMENT_VARIABLES.HUBSPOT_PORTAL_ID];
+  const refreshToken = process.env[ENVIRONMENT_VARIABLES.HUBSPOT_REFRESH_TOKEN];
+  const hubspotEnvironment =
+    process.env[ENVIRONMENT_VARIABLES.HUBSPOT_ENVIRONMENT];
+  const httpTimeoutVar = process.env[ENVIRONMENT_VARIABLES.HTTP_TIMEOUT];
+  const httpUseLocalhostVar =
+    process.env[ENVIRONMENT_VARIABLES.HTTP_USE_LOCALHOST];
+  const allowUsageTrackingVar =
+    process.env[ENVIRONMENT_VARIABLES.ALLOW_USAGE_TRACKING];
+  const defaultCmsPublishModeVar =
+    process.env[ENVIRONMENT_VARIABLES.DEFAULT_CMS_PUBLISH_MODE];
+
+  if (!accountIdVar) {
+    throw new Error(
+      i18n('config.utils.buildConfigFromEnvironment.missingAccountId')
+    );
+  }
+
+  const accountId = parseInt(accountIdVar);
+  const httpTimeout = httpTimeoutVar ? parseInt(httpTimeoutVar) : undefined;
+  const httpUseLocalhost = httpUseLocalhostVar
+    ? httpUseLocalhostVar === 'true'
+    : undefined;
+  const allowUsageTracking = allowUsageTrackingVar
+    ? allowUsageTrackingVar === 'true'
+    : undefined;
+  const defaultCmsPublishMode =
+    defaultCmsPublishModeVar === CMS_PUBLISH_MODE.draft ||
+    defaultCmsPublishModeVar === CMS_PUBLISH_MODE.publish
+      ? defaultCmsPublishModeVar
+      : undefined;
+
+  const env = getValidEnv(hubspotEnvironment);
+
+  let account: HubSpotConfigAccount;
+
+  if (personalAccessKey) {
+    account = {
+      authType: PERSONAL_ACCESS_KEY_AUTH_METHOD.value,
+      accountId,
+      personalAccessKey,
+      env,
+      name: accountIdVar,
+      auth: {
+        tokenInfo: {},
+      },
+    };
+  } else if (clientId && clientSecret && refreshToken) {
+    account = {
+      authType: OAUTH_AUTH_METHOD.value,
+      accountId,
+      auth: {
+        clientId,
+        clientSecret,
+        scopes: OAUTH_SCOPES.map((scope: { value: string }) => scope.value),
+        tokenInfo: {
+          refreshToken,
+        },
+      },
+      env,
+      name: accountIdVar,
+    };
+  } else if (apiKey) {
+    account = {
+      authType: API_KEY_AUTH_METHOD.value,
+      accountId,
+      apiKey,
+      env,
+      name: accountIdVar,
+    };
+  } else {
+    throw new Error(
+      i18n('config.utils.buildConfigFromEnvironment.invalidAuthType')
+    );
+  }
+
+  return {
+    accounts: [account],
+    defaultAccount: accountId,
+    httpTimeout,
+    httpUseLocalhost,
+    allowUsageTracking,
+    defaultCmsPublishMode,
+  };
+}
+
+export function getAccountIdentifierAndType(
+  accountIdentifier: string | number
+): { identifier: string | number; identifierType: 'name' | 'accountId' } {
+  const identifierAsNumber =
+    typeof accountIdentifier === 'number'
+      ? accountIdentifier
+      : parseInt(accountIdentifier);
+  const isId = !isNaN(identifierAsNumber);
+
+  return {
+    identifier: isId ? identifierAsNumber : accountIdentifier,
+    identifierType: isId ? 'accountId' : 'name',
```

Comment:
Not a blocker, but could it potentially prevent some future pain by making these identifiers constants?